### PR TITLE
Use Schedule::restart_groups()

### PR DESCRIPTION
**Pull request by joakim-hove**
Actually use the `Schedule::restart_groups()`- this should have been part of #1838

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
Opinions on this?

**Comment by atgeirr** [comment]
> Opinions on this?

I think maybe I am not the one you need an opinion from, I just have a question: what is a restart group?

**Comment by joakim-hove** [comment]
> I just have a question: what is a restart group?

It is a vector of normal groups which are ordered in a way suitable for writing restart files. I will add a comment  and merge it.

File: tests/parser/ScheduleTests.cpp [hunk 1]
```diff
@@ -258,6 +258,8 @@ static std::string createDeckWithWellsOrdered() {
     std::string input =
             "START             -- 0 \n"
             "10 MAI 2007 / \n"
+            "WELLDIMS\n"
+            "   *  *   3 /\n"
```
Comment (bska) [inline tests/parser/ScheduleTests.cpp:262]: Tiny nit, these should probably be
```
1*
```
instead of bare asterisks.

**Pull request merged**